### PR TITLE
[automativeskill] MainDialog to waterfall

### PR DESCRIPTION
**Pull request by xhr0804**
### Purpose
*What is the context of this pull request? Why is it being done?*
Change MainDialog of automativeskill to Waterfall 

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [ ] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

File: skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs [hunk 1]
```diff
@@ -1,8 +1,4 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (bobokids) [inline skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs]: Keep the file header

File: skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs [hunk 2]
```diff
@@ -12,70 +8,185 @@
...
             else
             {
-                var result = await luisService.RecognizeAsync<Luis.SettingsLuis>(dc.Context, CancellationToken.None);
-                var intent = result?.TopIntent().intent;
+                // If bot is in local mode, prompt with intro or continuation message
+                var prompt = stepContext.Options as Activity ?? _responseManager.GetResponse(AutomotiveSkillMainResponses.FirstPromptMessage);
+                var state = await _stateAccessor.GetAsync(stepContext.Context, () => new AutomotiveSkillState());
+                if (state.NewConversation)
+                {
+                    prompt = _responseManager.GetResponse(AutomotiveSkillMainResponses.WelcomeMessage);
```
Comment (KayMKM) [inline skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs]: what's the difference between FirstPromptMessage and WelcomeMessage? As my understanding, they should be same in intro step.

File: skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs [hunk 3]
```diff
@@ -84,98 +195,46 @@ protected override async Task OnMessageActivityAsync(DialogContext dc, Cancellat
...
-                                }
-                        }
-                    }
-                }
+                await stepContext.Context.SendActivityAsync(endOfConversation, cancellationToken);
+                return await stepContext.EndDialogAsync();
+            }
+            else
+            {
+                return await stepContext.ReplaceDialogAsync(this.Id, _responseManager.GetResponse(AutomotiveSkillMainResponses.WelcomeMessage), cancellationToken);
```
Comment (KayMKM) [inline skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs]: Need the WelcomeMessage be replaced by CompletedMessage?

File: skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs [hunk 4]
```diff
@@ -12,70 +8,185 @@
...
-                throw new Exception("The specified LUIS Model could not be found in your Bot Services configuration.");
+                // If the bot is in skill mode, skip directly to route and do not prompt
+                return await stepContext.NextAsync();
             }
             else
             {
-                var result = await luisService.RecognizeAsync<Luis.SettingsLuis>(dc.Context, CancellationToken.None);
-                var intent = result?.TopIntent().intent;
+                // If bot is in local mode, prompt with intro or continuation message
+                var prompt = stepContext.Options as Activity ?? _responseManager.GetResponse(AutomotiveSkillMainResponses.FirstPromptMessage);
```
Comment (KayMKM) [inline skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs]: Why don't use WelcomeMessage here?

File: skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs (hunk 3 above)
Comment (xhr0804) [reply skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs]: WelcomeMessage will only be used when state.NewConversation == true below

File: skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs (hunk 4 above)
Comment (xhr0804) [reply skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs]: WelcomeMessage will only be used when state.NewConversation == true below

File: skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs (hunk 1 above)
Comment (xhr0804) [reply skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs]: fixed

File: skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs (hunk 3 above)
Comment (xhr0804) [reply skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs]: sure, fixed

**Review by darrenj — APPROVED** [review]

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**